lib/Canonicalize/SimplifySwitchPass.cpp

```diff
@@ -0,0 +1,263 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+// Without this Pass the devirtualization process effectively handles indirect
+// jumps, particularly those arising from C's switch statement. However, the
+// output is too low-level, especially when switch statements involve jump
+// tables. This results in emitted memory accesses instead of clear switch
+// statements based on variable values. For instance, rather than a readable
+// switch statement reflecting _argument_rdi values, the code directly accesses
+// the jump table's entries. To address this readability issue, we propose
+// introducing the SimplifySwtichPass. This pass should refine the output of
+// switch statements by re-running the devirtualization process and adjusting
+// them to a more human-readable format. This enhancement is crucial for
+// improving code comprehension.
+
+#include "llvm/Passes/PassBuilder.h"
+
+#include "revng/Lift/LoadBinaryPass.h"
+#include "revng/Pipeline/RegisterPipe.h"
+#include "revng/Pipes/FileContainer.h"
+#include "revng/ValueMaterializer/ValueMaterializer.h"
+
+#include "revng-c/Support/IRHelpers.h"
+
+static Logger<> Log("switch-opt");
```

Comment:
Add `using namespace llvm` and drop all the `llvm::`.

Reply:
Not fixed

Reply:
I have not pushed the code back then, but I marked all the comments as resolved...